### PR TITLE
Chore: Adds Swagger `@Tag` annotations to API controllers

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

This PR adds the `@Tag` annotation to the REST API Controllers to enable better rendering in HTML

## Why it does that

To improve readability.

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Review by florianrusch-zf — APPROVED** [review]

File: extensions/api/data-management/asset/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/asset/AssetApi.java [hunk 1]
```diff
@@ -15,14 +15,15 @@
...
+import io.swagger.v3.oas.annotations.tags.Tag;
 import org.eclipse.dataspaceconnector.api.datamanagement.asset.model.AssetDto;
 import org.eclipse.dataspaceconnector.api.datamanagement.asset.model.AssetEntryDto;
-import org.eclipse.dataspaceconnector.api.datamanagement.asset.model.TransferRequestDto;
 import org.eclipse.dataspaceconnector.spi.query.SortOrder;
 
 import java.util.List;
 
 @OpenAPIDefinition
+@Tag(name = "Assets")
```
Comment (ndr-brt) [inline extensions/api/data-management/asset/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/asset/AssetApi.java:26]: That's the only written in plural form
```suggestion
@Tag(name = "Asset")
```

File: extensions/api/observability/src/main/java/org/eclipse/dataspaceconnector/api/observability/ObservabilityApiController.java [hunk 2]
```diff
@@ -29,6 +30,7 @@
 @Consumes({ MediaType.APPLICATION_JSON })
 @Produces({ MediaType.APPLICATION_JSON })
 @Path("/check")
+@Tag(name = "Application Observability")
```
Comment (ndr-brt) [inline extensions/api/observability/src/main/java/org/eclipse/dataspaceconnector/api/observability/ObservabilityApiController.java:33]: We agreed on putting the swagger annotations on the `*Api` interface, not on the `*ApiController` implementation

File: extensions/data-plane-selector/selector-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/selector/api/DataplaneSelectorApiController.java [hunk 3]
```diff
@@ -28,6 +29,7 @@
 @Consumes({ MediaType.APPLICATION_JSON })
 @Produces({ MediaType.APPLICATION_JSON })
 @Path("/instances")
+@Tag(name = "Dataplane Selector")
```
Comment (ndr-brt) [inline extensions/data-plane-selector/selector-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/selector/api/DataplaneSelectorApiController.java:32]: We agreed on putting the swagger annotations on the `*Api` interface, not on the `*ApiController` implementation. In this case we need to extract the interface as well

File: extensions/http-provisioner/src/main/java/org/eclipse/dataspaceconnector/transfer/provision/http/webhook/HttpProvisionerWebhookApiController.java [hunk 4]
```diff
@@ -34,6 +35,7 @@
 @Consumes({ MediaType.APPLICATION_JSON })
 @Produces({ MediaType.APPLICATION_JSON })
 @Path("/callback")
+@Tag(name = "HTTP Provisioner Webhook")
```
Comment (ndr-brt) [inline extensions/http-provisioner/src/main/java/org/eclipse/dataspaceconnector/transfer/provision/http/webhook/HttpProvisionerWebhookApiController.java:38]: We agreed on putting the swagger annotations on the `*Api` interface, not on the `*ApiController` implementation.

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**